relay-dynamic-config/src/metrics.rs

```diff
@@ -170,13 +171,113 @@ impl TransactionMetricsConfig {
     }
 }
 
+/// Combined view of global and project-specific metrics extraction configs.
+#[derive(Debug, Clone)]
+pub struct CombinedMetricExtractionConfig<'a> {
```

Comment:
Technically, for historic reasons we would call this `DynamicMetricExtractionConfig`, although this is now the 3rd time where an independent person used `Combined` instead of `Dynamic` (me, Iker, you) and I think it's finally time to retire `Dynamic` and stick to `Combined`.

It started with `DynamicMeasurementsConfig`, then came `DynamicQuotas` and afterwards `DynamicGenericFilters*`.